### PR TITLE
EditEmployee: Populates option boxes with current information and add…

### PR DESCRIPTION
**Pull request by ACast1**
…ed resave_cache function

*resave_cache() does not seem to work as expected
AddEmployee: added resave_cache function
*resave_cache() does not seem to work as expected

### CODE REVIEW

**Comment by zacryol** [comment]
Yes, I see, I messed that up a bit; I'll take care of it

**Comment by ACast1** [comment]
Thank you, can I get your analysis on the data population for the drop down menus?

File: project/Add_Employee_Frame.py [hunk 1]
```diff
@@ -264,9 +264,8 @@ def save_info(self):
         self.controller.database.set_employee_data(e_id, 'email', self.add_email_entry.get())
         self.controller.database.set_employee_data(e_id, 'end_date', self.add_end_date_entry.get())
 
-        #If password box is not empty
-        if self.add_password_entry.get() != "":
```
Comment (zacryol) [inline project/Add_Employee_Frame.py:268]: Why was the `if` removed?
That made sense to me to have there; make sure that the password is explicitly changed, and not require the employee to type it in each time or accidentally change it.

File: project/Edit_Employee_Frame.py [hunk 2]
```diff
@@ -30,6 +35,37 @@ def update(self):
             #Set employee to present as the currently logged in user
             self.e_id_to_present = self.controller.app_data["LoginFrame_userID"]
 
+        #If employee payment method is ACH
+        if str(self.controller.database.get_employee_data(self.e_id_to_present, "pay_method")) == "1":
```
Comment (zacryol) [inline project/Edit_Employee_Frame.py:39]: This bit could probably use an int cast to simplify a lot
something like 
```python
pay_method_value = int(self.controller_database.get_employee_data(self.e_id_to_present, "pay_method"))
self.pay_method.set(self.pay_method_options[pay_method_value]
```
you would need to ensure that the tuple order corresponds with the method/classification that each number represents.

File: project/Add_Employee_Frame.py (hunk 1 above)
Comment (zacryol) [reply project/Add_Employee_Frame.py:268]: Oh wait nevermind, this is creating the password, isn't it.

Comment (ACast1) [reply project/Add_Employee_Frame.py:268]: Correct

**Pull request merged**